diff --git a/ctdb/wscript b/ctdb/wscript
index b826b65ceb5..511edcf1d24 100644
--- a/ctdb/wscript
+++ b/ctdb/wscript
@@ -821,7 +821,7 @@ def build(bld):
     bld.INSTALL_FILES("${CTDB_HELPER_BINDIR}", "statd_callout_helper",
                       destname="statd_callout_helper", chmod=MODE_755)

-    bld.symlink_as(os.path.join(bld.env.CTDB_ETCDIR, "statd-callout"),
+    bld.symlink_as(os.path.join('${EXEC_PREFIX}${CTDB_ETCDIR}', "statd-callout"),
                    os.path.join(bld.env.CTDB_HELPER_BINDIR, "statd_callout"))

     def SUBDIR_MODE_callback(arg, dirname, fnames):
@@ -854,7 +854,7 @@ def build(bld):
         configdir = 'ctdb/config'

     for t in event_script_subdirs:
-        bld.INSTALL_DIR(os.path.join(bld.env.CTDB_ETCDIR, t))
+        bld.INSTALL_DIR(os.path.join('${EXEC_PREFIX}${CTDB_ETCDIR}', t))
         files = SUBDIR_MODE('%s/%s' % (configdir, t), trim_path=configdir)
         for fmode in files:
             bld.INSTALL_FILES(bld.env.CTDB_DATADIR, 'config/%s' % fmode[0],
@@ -871,7 +871,7 @@ def build(bld):
     if os.environ.get('DESTDIR') is None:
         fmt = 'events/legacy/%s.script'
         required_script = '00.ctdb'
-        required_path = os.path.join(bld.env.CTDB_ETCDIR,
+        required_path = os.path.join('${EXEC_PREFIX}${CTDB_ETCDIR}',
                                      fmt % (required_script))
         if not os.path.islink(required_path) and \
            not os.path.exists(required_path):
@@ -883,14 +883,14 @@ def build(bld):
                               ]
             for t in default_scripts:
                 tgt = os.path.join(bld.env.CTDB_DATADIR, fmt % (t))
-                name = os.path.join(bld.env.CTDB_ETCDIR, fmt % (t))
+                name = os.path.join('${EXEC_PREFIX}${CTDB_ETCDIR}', fmt % (t))
                 bld.symlink_as(name, tgt)

     bld.SAMBA_GENERATOR('ctdb-functions',
                         source='config/functions',
                         target='functions',
                         rule='sed %s ${SRC} > ${TGT}' % (sed_cmdline))
-    bld.INSTALL_FILES(bld.env.CTDB_ETCDIR, 'functions', destname='functions')
+    bld.INSTALL_FILES('${EXEC_PREFIX}${CTDB_ETCDIR}', 'functions', destname='functions')

     etc_scripts = [
         'ctdb-backup-persistent-tdbs.sh',
@@ -905,13 +905,10 @@ def build(bld):
         bld.INSTALL_FILES('${EXEC_PREFIX}${CTDB_ETCDIR}', 'config/%s' % t,
                           destname=t, chmod=MODE_755)

-    bld.INSTALL_FILES('${EXEC_PREFIX}$${CTDB_ETCDIR}/events/notification',
+    bld.INSTALL_FILES('${EXEC_PREFIX}${CTDB_ETCDIR}/events/notification',
                       'config/notification.README',
                       destname='README')

-    for d in ['volatile', 'persistent', 'state']:
-        bld.INSTALL_DIR(os.path.join(bld.env.CTDB_VARDIR, d))
-
     #
     # Test-only below this point
     #
